gui/wxpython/startup/guiutils.py

```diff
@@ -139,6 +148,107 @@ def _locationAlreadyExists(self, ctrl):
+        messages += get_reasons_mapset_not_removable(grassdb, location,
+                                                     mapset, check_permanent)
+    return messages
+
+
+def get_reasons_mapset_not_removable(grassdb, location, mapset, check_permanent):
+    """Get reasons why one mapset cannot be removed.
+    Parameter *check_permanent* is True of False. It depends on whether
+    we want to check for permanent mapset or not.
+
+    Returns messages as list if there were any failed checks or None whether not.
+    """
+    messages = []
+    mapset_path = os.path.join(grassdb, location, mapset)
+
+    # Check if mapset is permanent
+    if check_permanent and mapset == "PERMANENT":
+        messages.append(_("Mapset <{mapset}> is required for a valid location.").format(
+            mapset=mapset_path))
+    # Check if mapset is current
+    elif is_mapset_current(grassdb, location, mapset):
+        messages.append(_("Mapset <{mapset}> is the current mapset.").format(
+            mapset=mapset_path))
+    # Check whether mapset is in use
+    elif is_mapset_locked(mapset_path):
+        messages.append(_("Mapset <{mapset}> is in use.").format(
+            mapset=mapset_path))
+    # Check whether mapset is owned by different user
+    elif is_different_mapset_owner(mapset_path):
+        messages.append(_("Mapset <{mapset}> is owned by a different user.").format(
+            mapset=mapset_path))
+
+    return messages
+
+
+def get_reasons_locations_not_removable(locations):
+    """Get reasons why locations cannot be removed.
+    Parameter *locations* is a list of tuples (database, location).
+    This is currently just a convenience wrapper for get_reasons_location_not_removable().
+    """
+    messages = []
+    for grassdb, location in locations:
+        messages += get_reasons_location_not_removable(grassdb, location)
+    return messages
+
+
+def get_reasons_location_not_removable(grassdb, location):
+    """Get reasons why one location cannot be removed.
+
+    Returns messages as list if there were any failed checks or None whether not.
+    """
+    messages = []
+    location_path = os.path.join(grassdb, location)
+
+    # Check if location is current
+    if is_location_current(grassdb, location):
+        messages.append(_("Location <{location}> is the current location.").format(
+            location=location_path))
+        return messages
+
+    # Find mapsets in particular location
+    tmp_gisrc_file, env = gs.create_environment(grassdb, location, 'PERMANENT')
+    env['GRASS_SKIP_MAPSET_OWNER_CHECK'] = '1'
+    maps_dict = {}
+    try:
+        g_mapsets = gs.read_command(
+            'g.mapsets',
+            flags='l',
+            separator='comma',
+            quiet=True,
+            env=env).strip().split(',')
+    except CalledModuleError:
+        Queue().put(
+            (maps_dict,
+             _("Failed to read mapsets from location <{location}>.").format(
+                 location=location_path)))
```

Comment:
You can't just blindly copy code. This is handling communication from parallelized calls of getLocationTree function, remove it.